finesse/hardware/__init__.py

```diff
@@ -10,31 +11,102 @@
     from .em27_scraper import EM27Scraper  # type: ignore
     from .opus.em27 import OPUSInterface  # type: ignore
 
-from . import data_file_writer  # noqa
-from .stepper_motor import create_stepper_motor_serial_manager
-from .temperature import (
-    create_temperature_controller_serial_managers,
-    create_temperature_monitor_serial_manager,
-)
+from datetime import datetime
 
-opus: OPUSInterface
+from finesse.config import NUM_TEMPERATURE_MONITOR_CHANNELS, TEMPERATURE_MONITOR_TOPIC
+from finesse.device_info import DeviceBaseTypeInfo, DeviceInstanceRef, DeviceTypeInfo
+
+from . import data_file_writer  # noqa: F401
+from .plugins import load_device_types
+from .plugins.temperature import get_temperature_monitor_instance
```

Comment:
Why is the temperature monitor treaded in a different way?

Reply:
It's a hack -- I mentioned it briefly in the PR description.

The temperatures are requested from the temperature monitor every 2s and they are used to a) update the plot and b) save into the data file (if recording is happening). The problem is that at some point Jon decided that he wanted data files to carry on recording regardless of whether the temperature monitor is active. In general this is a better design, because it makes things more resilient in the case of the temperature monitor breaking, but the way I implemented it was to add a hack that NaNs are broadcast in place of real temperatures if the device isn't working. A better way to do it would be to have all the sensors whose data *can* be recorded into data files respond to a different, specific message used for this purpose only.

I've opened a new issue and copied this description into it: #337